org.eclipse.jface.text/src/org/eclipse/jface/text/ITextViewerLifecycle.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * - Angelo ZERR (Red Hat Inc.)  - initial implementation
+ *******************************************************************************/
+package org.eclipse.jface.text;
+
+/**
+ * {@link ITextViewer} lifecycle API to track install / uninstall of a given {@link ITextViewer}.
+ *
+ * @since 3.21.0
+ *
+ */
+public interface ITextViewerLifecycle {
```

Comment:
After having read the JavaDoc I wonder where this API is being used. It doesn’t look like any check for the newly introduced interface happens anywhere. Naively I might assume that a random new service instance returned from e.g. a ISourceViewerConfig would be checked and the install would be called. But in fact this is not happening. Please add to the docs what an implementor can expect from adding this supertype and which services are allowed to be ITextViewerLifecycles

Reply:
Ok I understand what you mean. My main motivation is to use ITextViewerLifecycle in generic editor https://github.com/eclipse-platform/eclipse.platform.text/issues/85 to track install / uninstall viewer inside generic editor contribution (auto editing stratetgy,etc). 

When I wanted to provide this API inside generic editor, I noticed that several API used install / unisntall methods with the same signature, so instead of providing this API in generic editor, I provide this API in JFace text directly.

In other words, this API doesn't provide a given service but it exists just to standardize the method name when you need to track install / uninstall.

Reply:
I think more and more that ItextViewerLyfecicle https://github.com/eclipse-platform/eclipse.platform.text/pull/107 should be hosted in generic editor

@mickaelistria @szarnekow what do you think about that?

Reply:
To me this would be reasonable. Maybe also rename to IGenericTextViewerLifecycle or something more suited?

Reply:
@szarnekow I move ITextViewerLifecycle to genereic editor and I keep the name ITextViewerLifecycle

Reply:
@szarnekow I moved again ITextViewerLifecycle in JFace Text by foolowing great suggestion of @mickaelistria.

Now this API should be relevant. I updated the Javadoc to explain that. Hope you will like it.